AirCasting/SDSync/ViewsFlow/SDSyncRootView/SDSyncRootView.swift

```diff
@@ -0,0 +1,85 @@
+// Created by Lunar on 16/11/2021.
+//
+
+import SwiftUI
+
+struct SDSyncRootView<VM: SDSyncRootViewModel>: View {
+    @StateObject var viewModel: VM
+    @EnvironmentObject private var finishAndSyncButtonTapped: FinishAndSyncButtonTapped
+    @Binding var creatingSessionFlowContinues: Bool
+    
+    var body: some View {
+        VStack(spacing: 40) {
+            ProgressView(value: 0.1)
+            Spacer()
+            ZStack(alignment: Alignment(horizontal: .trailing, vertical: .bottom), content: {
+                syncImage
+                loader
+                    .padding()
+                    .padding(.vertical)
+            })
+            Spacer()
+            VStack(alignment: .leading, spacing: 15) {
+                titleLabel
+                messageLabel
+            }
+            Spacer()
+        }
+        .padding()
+        .background(navigationLink)
+        .onAppear() {
+            viewModel.onAppearExecute()
+            finishAndSyncButtonTapped.finishAndSyncButtonWasTapped = false
+            viewModel.startBackendSync()
```

Comment:
I don't understand the logic here. Why are we calling onAppearExecute, which is checking if there is a sync in progress, and if there is, it is calling startBackendSync when it ends and then we are calling the stantBackedSync again from here? As I understand, it could happen that we would call sync twice?